src/angular-app/bellows/shared/audio-recorder/audio-recorder.component.ts

```diff
@@ -1,163 +1,126 @@
+    .then(stream => {
+
+      this.mediaRecorder = new MediaRecorder(stream);
 
       this.$scope.$apply(() => {
         this.hasRecorded = true;
         this.errorMessage = null;
         this.isRecording = true;
       });
 
-      const recordingStartTime = new Date();
-      // webkit prefix required for Safari
-      const context = (window as any).AudioContext ? new AudioContext() : new webkitAudioContext();
-      const bufferSize = 0;
-      const channels = 1;
-      const processor = context.createScriptProcessor(bufferSize, channels, channels);
-      context.createMediaStreamSource(stream).connect(processor);
-      processor.connect(context.destination);
-      const sampleRate = context.sampleRate;
-      const bitrate = 128;
-      const mp3Encoder = new MP3Encoder(channels, sampleRate, bitrate);
-
-      function handleAudioData(event: AudioProcessingEvent) {
-        mp3Encoder.appendData(event.inputBuffer.getChannelData(0));
-      }
-      processor.addEventListener('audioprocess', handleAudioData);
-
-      mp3Encoder.onMp3Blob(blob => {
-        this.blob = blob;
-        this.audioSrc = URL.createObjectURL(blob);
+
+      this.mediaRecorder.addEventListener('dataavailable', async (e: { data: any; }) => {
+        this.chunks.push(e.data);
+        var roughBlob = new Blob(this.chunks, {type: 'audio/webm; codecs=opus'});
+        //2022-09
+        //In some browsers (Chrome, Edge, ...) navigator.mediaDevices.getUserMedia with MediaRecorder creates WEBM files without duration metadata
+        //webmFixDuration appends missing duration metadata to a WEBM file blob.
+        this.blob = await webmFixDuration(roughBlob, this.durationInMilliseconds, 'audio/webm; codecs=opus');
+        this.chunks = [];
+        this.audioSrc = window.URL.createObjectURL(this.blob);
+        this.$scope.$digest();
       });
 
+      this.recordingStartTime = new Date();
+
       this.interval = this.$interval(() => {
-        const seconds = Math.floor((new Date().getTime() - recordingStartTime.getTime()) / 1000);
+        const seconds = Math.floor((new Date().getTime() - this.recordingStartTime.getTime()) / 1000);
         this.recordingTime = Math.floor(seconds / 60) + ':' + (seconds % 60 < 10 ? '0' : '') + seconds % 60;
       }, 1000);
 
-      this.stopMediaStream = () => {
-        processor.removeEventListener('audioprocess', handleAudioData);
-        mp3Encoder.end();
-        stream.getAudioTracks()[0].stop();
-      };
+
+      this.mediaRecorder.start();
 
     }, err => {
+
       this.$scope.$apply(() => {
         this.errorMessage = 'Unable to record audio from your microphone.';
         this.isRecording = false;
         this.hasRecorded = false;
       });
-      console.error(err);
-    });
-  }
 
-  private stopRecording() {
-    if (this.interval) this.$interval.cancel(this.interval);
-    if (this.stopMediaStream) this.stopMediaStream();
-  }
+      console.error(err);
 
-}
+      });//END getUserMedia call
```

Comment:
nit: space between semi colon.  I think we need Prettier installed to help with this.  Can you install the Prettier VSCode extension?

Reply:
Thanks. Installed Prettier and made that change. I'm also taking out a couple unneeded comments that I see I left in this file.